src/client/auth/aws.rs

```diff
@@ -75,7 +72,8 @@ async fn authenticate_stream_inner(
         // channel binding is not supported.
         "p": 110i32,
     };
-    let client_first_payload_bytes = client_first_payload.encode_to_vec()?;
+    let mut client_first_payload_bytes = vec![];
```

Comment:
I don't think the few places `to_vec` would be used is worth keeping the extra compat overhead for it.